src/main/java/jenkins/scm/api/SCMNavigatorDescriptor.java

```diff
@@ -84,7 +84,9 @@ public String getIconFilePathPattern() {
     /**
      * Creates a default navigator, if there should be one displayed.
      * May be interpreted by {@code jenkins.branch.SpecificDescriptor}.
-     * If returning non-null, you may also provide a {@code newInstanceDetail} view as per {@link TopLevelItemDescriptor}.
+     * If returning non-null, you may also provide a {@code newInstanceDetail} view as per {@link TopLevelItemDescriptor}
+     * and implement {@link SCMNavigatorDescriptor#getDescription()}.
```

Comment:
And it might be worth pointing out in the three new methods that they will not be used unless you return something from `newInstance`. In other words, if leaving `newInstance` to return null, you need not bother overriding the new methods.